agent/util-scripts/pbench-tool-meister-start

```diff
@@ -0,0 +1,450 @@
+            continue
+        else:
+            if new_data["kind"] == "ds":
+                pids["ds"] = new_data
+                have_ds = True
+            elif new_data["kind"] == "tm":
+                if "tm" not in pids:
+                    pids["tm"] = []
+                pids["tm"].append(new_data)
+                num_tms += 1
+            else:
+                logger.warning("unrecognized 'kind', in data payload '%r'", data)
+                continue
+        if have_ds and num_tms == expected_tms:
+            break
+    # Record our collected pids.
+    redis_server.set("tm-pids", json.dumps(pids))
+
+
+def main(argv):
+    """Main program for the tool meister start.
+    """
+    PROG = os.path.basename(argv[0])
+    logger = logging.getLogger(PROG)
+    logger.setLevel(logging.DEBUG)
+    sh = logging.StreamHandler()
+    sh.setLevel(logging.DEBUG)
+    shf = logging.Formatter("%(message)s")
+    sh.setFormatter(shf)
+    logger.addHandler(sh)
+
+    try:
+        group = argv[1]
+    except IndexError:
+        group = "default"
+
+    # 1. Load the tool group data given the tool group argument
+    try:
+        tool_group = ToolGroup(group)
+    except Exception:
+        logger.exception("failed to load tool group data")
+        return 1
+
+    try:
+        benchmark_run_dir = os.environ["benchmark_run_dir"]
+        hostname = os.environ["hostname"]
+        full_hostname = os.environ["full_hostname"]
+    except Exception:
+        logger.exception("failed to fetch parameters from the environment")
+        return 1
+    else:
+        tm_dir = os.path.join(benchmark_run_dir, "tm")
+        try:
+            os.mkdir(tm_dir)
+            os.chdir(tm_dir)
+        except Exception:
+            logger.exception("failed to create the local tool meister" " directory")
+            return 1
+    if not full_hostname or not hostname:
+        logger.error(
+            "ERROR - hostname ('%s') and full_hostname ('%s') environment"
+            " variables are required",
+            hostname,
+            full_hostname,
+        )
+    if os.environ.get("_PBENCH_UNIT_TESTS"):
+        # FIXME: this is an artifact of the unit test environment.
+        hostnames = "localhost"
+    else:
+        hostnames = f"localhost {full_hostname}"
+    params = {"hostnames": hostnames, "tm_dir": tm_dir, "redis_port": redis_port}
+
+    # 2. Start the Redis Server (config of port from agent config)
+    #   - the Redis Server is requested to create the PID file
+
+    # Create the Redis Server pbench-specific configuration file
+    redis_conf = os.path.join(tm_dir, "redis.conf")
+    try:
+        with open(redis_conf, "w") as fp:
+            fp.write(redis_conf_tmpl.format(**params))
+    except Exception:
+        logger.exception("failed to create redis server configuration")
+        return 1
+    # Start the Redis Server itself
+    #   - FIXME: use podman to start a redis server container
+    redis_srvr = "redis-server"
```

Comment:
What if the user is already running a redis-server?

Reply:
We don't support that with this commit.  In the future we might.